### PR TITLE
[WIP][NOTEST][RFR][1LP] Sprout - Able to select a specific provider type

### PR DESCRIPTION
**Pull request by mfalesni**
@lcouzens ?

### CODE REVIEW

**Comment by lcouzens** [comment]
This looks great, question is it just for requesting appliances through sprout? or is it also usable with sprout_client.provision_appliances?

**Comment by mfalesni** [comment]
@lcouzens I added `provider_type` into the api call as well, so after modification of the sprout client afterwards, it should work via api.

**Comment by lcouzens** [comment]
Awesome, nice work man!

File: sprout/appliances/models.py [hunk 1]
```diff
@@ -1180,8 +1204,15 @@ def create(cls, owner, group, version=None, date=None, provider=None, num_applia
...
         if isinstance(group, basestring):
             group = Group.objects.get(id=group)
+        if isinstance(provider_type, basestring):
+            if provider_type not in Provider.get_available_provider_types(owner):
+                raise Exception('There are no providers for type {!r}'.format(provider_type))
         if isinstance(provider, basestring):
             provider = Provider.objects.get(id=provider, working=True, disabled=False)
+            if provider_type is not None and provider.provider_type != provider_type:
+                raise Exception(
+                    'You used contradicting parameters: provider={!r}, provider_type={!r}'.format(
```
Comment (mshriver) [inline sprout/appliances/models.py]: :+1:

File: sprout/appliances/models.py [hunk 2]
```diff
@@ -1180,8 +1204,15 @@ def create(cls, owner, group, version=None, date=None, provider=None, num_applia
                 date = dates[0]
         if isinstance(group, basestring):
             group = Group.objects.get(id=group)
+        if isinstance(provider_type, basestring):
+            if provider_type not in Provider.get_available_provider_types(owner):
+                raise Exception('There are no providers for type {!r}'.format(provider_type))
```
Comment (mshriver) [inline sprout/appliances/models.py]: Consider ValueError instead of Exception for these two

**Pull request merged**